plugins/importexport/native/filter/NativeXmlSubmissionFileFilter.inc.php

```diff
@@ -145,197 +95,230 @@ function handleRevisionElement($node, $stageId, $fileId) {
-		$submissionFile = $submissionFileDao->newDataObjectByGenreId($genreId);
-		$submissionFile->setSubmissionId($submission->getId());
-		$submissionFile->setSubmissionLocale($submission->getLocale());
-		$submissionFile->setGenreId($genreId);
-		$submissionFile->setFileStage($stageId);
-		$submissionFile->setDateUploaded(Core::getCurrentDate());
-		$submissionFile->setDateModified(Core::getCurrentDate());
-		if ($node->getAttribute('available') == 'true') $submissionFile->setViewable(true);
-
-		$submissionFile->setOriginalFileName($filename = $node->getAttribute('filename'));
-		for ($n = $node->firstChild; $n !== null; $n=$n->nextSibling) {
-			if (is_a($n, 'DOMElement')) {
-				$filename = $this->handleRevisionChildElement($n, $submission, $submissionFile);
-			}
-		}
-		if (!$filename) {
-			// $this->handleRevisionChildElement() failed to provide any file (error message should have been handled in called method)
-			$errorOccured = true;
-		} else {
-			clearstatcache(true, $filename);
-			if (!file_exists($filename) || !filesize($filename)) {
-				// $this->handleRevisionChildElement() failed to provide a real file
-				$deployment->addError(ASSOC_TYPE_SUBMISSION, $submission->getId(), __('plugins.importexport.native.error.submissionFileImportFailed'));
-				$errorOccured = true;
-			}
-		}
-
 		$uploaderUsername = $node->getAttribute('uploader');
+		$uploaderUserId = null;
 		if (!$uploaderUsername) {
 			$user = $deployment->getUser();
 		} else {
 			// Determine the user based on the username
 			$userDao = DAORegistry::getDAO('UserDAO'); /* @var $userDao UserDAO */
 			$user = $userDao->getByUsername($uploaderUsername);
 		}
-		if ($user) {
-			$submissionFile->setUploaderUserId($user->getId());
-		} else {
-			$deployment->addError(ASSOC_TYPE_SUBMISSION, $submission->getId(), __('plugins.importexport.common.error.unknownUploader', array('param' => $uploaderUsername)));
-			$errorOccured = true;
-		}
+		$uploaderUserId = $user
+			? (int) $user->getId()
+			: Application::get()->getRequest()->getUser()->getId();
 
-		$fileSize = $node->getAttribute('filesize');
-		$fileSizeOnDisk = filesize($filename);
-		if ($fileSize) {
-			if ($fileSize != $fileSizeOnDisk) {
-				$deployment->addWarning(ASSOC_TYPE_SUBMISSION, $submission->getId(), __('plugins.importexport.common.error.filesizeMismatch', array('expected' => $fileSize, 'actual' => $fileSizeOnDisk)));
-			}
+		$submissionFileDao = DAORegistry::getDAO('SubmissionFileDAO'); /* @var $submissionFileDao SubmissionFileDAO */
+		$submissionFile = $submissionFileDao->newDataObject();
+		$submissionFile->setData('submissionId', $submission->getId());
+		$submissionFile->setData('locale', $submission->getLocale());
+		$submissionFile->setData('fileStage', $stageId);
+		$submissionFile->setData('createdAt', Core::getCurrentDate());
+		$submissionFile->setData('updatedAt', Core::getCurrentDate());
```

Comment:
@asmecher, when importing a submission file I'm setting the `createdAt` and `updatedAt` dates to reflect the import itself as an "upload" and an "action". But I see that this is different from how it was before. Should I preserve the existing `createdAt` and `updatedAt` dates when importing?

Reply:
I'm not sure I'm following -- but looking at the previous code, the file dates were set to the current date as you've done here, which is good. I could foresee adding optional attributes at some point so these could be specified in the import document, but that's not a use case we've heard yet, so no need to prioritize.